contracts/mining.js

```diff
@@ -700,10 +712,25 @@ async function runLottery(pool, params) {
   let computedWinners = 0;
   const winners = [];
   while (computedWinners < pool.lotteryWinners) {
-    miningPowers = await api.db.find('miningPower', { id: pool.id, power: { $gt: { $numberDecimal: '0' } } },
-      params.processQueryLimit,
-      offset,
-      [{ index: 'power', descending: true }, { index: '_id', descending: false }]);
+    if (!pool.externalContract) {
```

Comment:
I wanted to have something like a defined method name to call in the external contract (i.e getMiningPower). Then we don't need clauses here for each contract we're hooking up to this. Other suggestions are welcome.